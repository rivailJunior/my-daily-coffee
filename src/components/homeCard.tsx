import Image from 'next/image';
import { Button } from './ui/button';
import { Link } from 'lucide-react';

interface HomeCardProps {
  onClick?: () => void;
  title: string;
  description: string;
  image: string;
}

export const HomeCard = ({
  onClick,
  title,
  description,
  image,
}: HomeCardProps) => {
  return (
    <div className=' bg-white border border-gray-200 rounded-lg shadow-sm dark:bg-gray-800 dark:border-gray-700'>
      <Image
        className='rounded-t-lg w-full h-48 md:h-80 object-cover'
        src={image}
        alt=''
        width={500}
        height={500}
      />
      <div className='p-5'>
        <a href='#'>
          <h5 className='mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white'>
            {title}
          </h5>
        </a>
        <p className='mb-3 font-normal text-gray-700 dark:text-gray-400'>
          {description}
        </p>
        <Button variant='default' onClick={onClick}>
          See more
        </Button>
      </div>
    </div>
  );
};

export const HomeCardHorizontal = ({
  onClick,
  title,
  description,
  image,
}: HomeCardProps) => {
  return (
    <div className='flex flex-row bg-white border border-gray-200 rounded-lg overflow-hidden shadow-sm md:flex-row w-full hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700 h-auto'>
      <Image
        className='object-cover w-20 h-20 md:h-auto md:w-20 md:rounded-none md:rounded-s-lg'
        src={image}
        alt=''
        width={500}
        height={500}
      />
      <div className='flex flex-col justify-around p-4 leading-normal'>
        <div className='text-lg font-medium tracking-tight text-gray-900 dark:text-white capitalize'>
          {title}
        </div>
        <div className='text-sm text-gray-700 dark:text-gray-400'>
          {description}
        </div>
      </div>
    </div>
  );
};