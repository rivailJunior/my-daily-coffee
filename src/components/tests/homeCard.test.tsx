import { render, screen, fireEvent } from '@testing-library/react';
import { HomeCard } from '../homeCard';

describe('HomeCard', () => {
  const mockOnClick = jest.fn();
  const defaultProps = {
    title: 'Test Title',
    description: 'Test Description',
    image: '/test-image.jpg',
    onClick: mockOnClick,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with required props', () => {
    render(<HomeCard {...defaultProps} />);

    expect(screen.getByText('Test Title')).toBeInTheDocument();
    expect(screen.getByText('Test Description')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src');
    expect(screen.getByText('See more')).toBeInTheDocument();
  });

  it('calls onClick when button is clicked', () => {
    render(<HomeCard {...defaultProps} />);

    const button = screen.getByText('See more');
    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { container } = render(<HomeCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
