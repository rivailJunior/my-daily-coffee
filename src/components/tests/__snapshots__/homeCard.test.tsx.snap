// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard matches snapshot 1`] = `
<div>
  <div
    class=" bg-white border border-gray-200 rounded-lg shadow-sm dark:bg-gray-800 dark:border-gray-700"
  >
    <img
      alt=""
      class="rounded-t-lg w-full h-48 md:h-80 object-cover"
      data-nimg="1"
      decoding="async"
      height="500"
      loading="lazy"
      src="/_next/image?url=%2Ftest-image.jpg&w=1080&q=75"
      srcset="/_next/image?url=%2Ftest-image.jpg&w=640&q=75 1x, /_next/image?url=%2Ftest-image.jpg&w=1080&q=75 2x"
      style="color: transparent;"
      width="500"
    />
    <div
      class="p-5"
    >
      <a
        href="#"
      >
        <h5
          class="mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
        >
          Test Title
        </h5>
      </a>
      <p
        class="mb-3 font-normal text-gray-700 dark:text-gray-400"
      >
        Test Description
      </p>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-coffee-navy hover:bg-coffee-navy/80 dark:bg-green-500 dark:hover:bg-green-500/80 text-white dark:text-white capitalize h-9 px-4 py-2"
      >
        See more
      </button>
    </div>
  </div>
</div>
`;
